kong/router/compat.lua

```diff
@@ -67,6 +72,68 @@ local LOGICAL_AND = atc.LOGICAL_AND
 local uuid_generator = assert(uuid.factory_v5('7f145bf9-0dce-4f91-98eb-debbce4b9f6b'))
 
 
+local function gen_for_nets(ip_field, port_field, vals)
+  if is_empty_field(vals) then
+    return nil
+  end
+
+  local nets_buf = buffer.new(64):put("(")
+
+  for i = 1, #vals do
+    local v = vals[i]
+
+    if type(v) ~= "table" then
+      ngx.log(ngx.ERR, "sources/destinations elements must be a table")
+      return nil
+    end
+
+    if is_empty_field(v) then
+      ngx.log(ngx.ERR, "sources/destinations elements must not be empty")
+      return nil
+    end
+
+    local ip = v.ip
+    local port = v.port
+
+    local exp_ip, exp_port
+
+    if ip then
+      local addr, mask = parse_ip_addr(ip)
+
+      if mask then  -- ip in cidr
+        exp_ip = ip_field .. " " .. OP_IN ..  " " ..
+                 addr .. "/" .. mask
+
+      else          -- ip == addr
+        exp_ip = ip_field .. " " .. OP_EQUAL .. " " ..
+                 addr
+      end
+    end
+
+    if port then
+      exp_port = port_field .. " " .. OP_EQUAL .. " " .. port
+    end
+
+    if not ip then
+      buffer_append(nets_buf, LOGICAL_OR, exp_port, i)
```

Comment:
Could you please make this function more readable, such as explaining the fourth parameter, renaming it to `expression_append`.

Reply:
Renaming `buffer_append` will cause many unnecessary diff in the PR, I will do it after this PR merged.